[WIP] Xenial support

jpic opened the pull request.

arBmind commented:
Please hold your breath the install for nine has changed quite a bit. I will push in an hour or so

jpic commented:
Thanks mate!! I'll be active on this on Monday morning any way, I'm in
UTC+2

arBmind commented:
@jpic  You should rebase your branch or it will not be mergeable. Base your work on my changes from friday - it should be a lot easier.

jpic commented:
I'm trying :)

jpic commented:
For ansible it works but the buildmaster won't start:

```
buildmaster@devops:~$ buildbot --verbose restart buildmaster
2016-08-08 14:31:06+0000 [-] Log opened.
2016-08-08 14:31:06+0000 [-] error reading '/home/buildmaster/buildmaster/buildbot.tac': No such file or directory
2016-08-08 14:31:06+0000 [-] invalid buildmaster directory '/home/buildmaster/buildmaster'
```


Playbook:

```
- name: setup master
  hosts: devops
  vars:
    buildbot_master_host: devops
  sudo: yes
  roles:
  - ansible-buildbot/master/install-nine
  - ansible-buildbot/master/create
  - ansible-buildbot/master/inplace-config
  - ansible-buildbot/project/add
  - ansible-buildbot/worker/create
  - ansible-buildbot/master/start-inplace
  - ansible-buildbot/worker/start
```

jpic commented:
Ok nvm should have used .

jpic commented:
That's the error i'm facing

```
2016-08-08 13:00:08+0000 [-] Loading buildbot.tac...
2016-08-08 13:00:09+0000 [-] Loaded.
2016-08-08 13:00:09+0000 [-] twistd 16.2.0 (/usr/bin/python 2.7.12) starting up.
2016-08-08 13:00:09+0000 [-] reactor class: twisted.internet.epollreactor.EPollReactor.
2016-08-08 13:00:09+0000 [-] Starting BuildMaster -- buildbot.version: 0.9.0
2016-08-08 13:00:09+0000 [-] Loading configuration from '/home/buildmaster/master.cfg'
2016-08-08 13:00:09+0000 [-] error while parsing config file:
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 246, in inContext
        result = inContext.theWork()
      File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 262, in <lambda>
        inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
      File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 118, in callWithContext
        return self.currentContext().callWithContext(ctx, func, *args, **kw)
      File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 81, in callWithContext
        return func(*args,**kw)
    --- <exception caught here> ---
      File "/opt/buildbot/master/buildbot/config.py", line 193, in loadConfig
        exec f in localDict
      File "/home/buildmaster/master.cfg", line 8, in <module>
        from buildbot_inplace import InplaceConfig
      File "inplace/buildbot_inplace/__init__.py", line 19, in <module>
        from config import Wrapper
      File "inplace/buildbot_inplace/config.py", line 24, in <module>
        from inplace_build import InplaceBuildFactory
      File "inplace/buildbot_inplace/inplace_build.py", line 22, in <module>
        from steps.checkout import create_checkout_step
      File "inplace/buildbot_inplace/steps/checkout.py", line 21, in <module>
        from .success import ShowStepIfSuccessful
      File "inplace/buildbot_inplace/steps/success.py", line 19, in <module>
        from buildbot.process.results import SUCCESS
    exceptions.ImportError: No module named results

2016-08-08 13:00:09+0000 [-] Configuration Errors:
2016-08-08 13:00:09+0000 [-]   error while parsing config file: No module named results (traceback in logfile)
```

jpic commented:
Created hicknhack-software/buildbot-inplace-config#1

arBmind commented:
That is strange... It worked here. Are you sure you have buildbot 0.9.0rc1 ?

jpic commented:
Buildbot version: 0.9.0

​

arBmind commented:
I took a look at the current changes: Why do we still need npm and gulp for Xenial? - It should not be necessary any more. We use the official buildbot releases, where all the assets are already compiled.

Why is your Xenial using Python2 - From my experience Xenial uses Python3 by default.

jpic commented:
I'm trying from your devel branch again without these patches, indeed it's
getting further than last week :)

Because we install the python package for ansible support, and the python
package refers to python2:
https://wiki.ubuntu.com/XenialXerus/ReleaseNotes#Python_3

jpic committed:
Add Xenial dependency for pyopenssl

jpic committed:
Add pyyaml pip dependency

Fixes

	Traceback (most recent call last):
	  File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 262, in <lambda>
	    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
	  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 118, in callWithContext
	    return self.currentContext().callWithContext(ctx, func, *args, **kw)
	  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 81, in callWithContext
	    return func(*args,**kw)
	  File "/usr/local/lib/python2.7/dist-packages/buildbot/config.py", line 151, in loadConfig
	    self.basedir, self.configFileName)
	--- <exception caught here> ---
	  File "/usr/local/lib/python2.7/dist-packages/buildbot/config.py", line 108, in loadConfigDict
	    exec(f, localDict)
	  File "/home/buildmaster/master.cfg", line 8, in <module>
	    from buildbot_inplace import InplaceConfig
	  File "inplace/buildbot_inplace/__init__.py", line 19, in <module>
	    from config import Wrapper
	  File "inplace/buildbot_inplace/config.py", line 24, in <module>
	    from inplace_build import InplaceBuildFactory
	  File "inplace/buildbot_inplace/inplace_build.py", line 24, in <module>
	    from steps.retrieve_inplace import RetrieveInplaceConfigStep
	  File "inplace/buildbot_inplace/steps/retrieve_inplace.py", line 22, in <module>
	    from ..inplace_config import InplaceConfig
	  File "inplace/buildbot_inplace/inplace_config.py", line 19, in <module>
	    from yaml import safe_load
	exceptions.ImportError: No module named yaml
